### DISCUSSION TITLE
Unterstützung von Gruppen schon implementiert?

### DISCUSSION THREAD

**Discussion by itn3rd77** [SukramJ/aiohomematic#28] [Q&A] [answered]
Hi zusammen,

vielen Dank erstmal @SukramJ  und @danielperna84. Großartig was ihr ihr leistet!

Habe eine Menge Geräte und würde auch gerne beim Testen helfen. Sind Gruppen eigentlich aktuell auch schon unterstützt? Dann würde ich nämlich mal "einsteigen".

Danke und viele Grüße
Ingo

**Comment by SukramJ** [comment, marked as answer]
Moin,

Hilfe beim Testen ist immer willkommen, und Climate-Gruppen werden schon unterstützt.
Das Projekt ist noch in einer seeehr frühen Phase und es gibt viele und teils grundlegende Änderungen.

Aktuell werden noch keine (Test-)Releases erstellt, so das es beim Update zu Problemen bei der Aktualisierung der Abhängigkeiten kommen kann, weswegen man manuell die hahhomematic dependency löschen muss.
Von der installation in HA-Supervised kann ich aktuell nur abraten, weil ich keine Ahnung hab wo die nachgeladenen Dependencies abgelegt werden.
Das Testen in einer VENV Umgebung ist aktuell die bessere Variante.

Ich hoffe das demnächst regelmäßig Test-Releases erstellt werden können. Der aktuelle Stand ist noch experimentell.

VG
Markus

**Reply by itn3rd77** [reply]
Danke für die schnelle Antwort Markus.

Wie sieht es den mit Updates über HACS aus? Das wäre für uns early adopter sicherlich der einfachste Weg...

**Reply by Lemocin** [reply]
Also bei mir erkennt HACS die neusten Versionen anhand der Commit-ID. Das updaten über HACS klappt hier problemlos. Probier es einfach mal

**Reply by emufan** [reply]
> Von der installation in HA-Supervised kann ich aktuell nur abraten, weil ich keine Ahnung hab wo die nachgeladenen Dependencies abgelegt werden.

@SukramJ Ist das noch der aktuelle Stand oder werden Dependencies auch in HA-Supervised abgelegt und vor allen Dingen aktualisiert? Und auch alles sauber gelöscht, wenn ich die Integration zwischendurch mal entferne, usw.?

**Reply by SukramJ** [reply]
Das ist mittlerweile kein Problem mehr, da wir jetzt Releases verwenden.

**Reply by emufan** [reply]
Danke!

**Comment by SukramJ** [comment]
Darum geht es ja.
Da custom_homematic und hahomematic aktuell noch nicht versioniert werden funktioniert zwar die Installation, aber beim Update wird keine neue hahomematic Version nachgeladen.

**Reply by itn3rd77** [reply]
OK. Got it 👍😉

**Reply by SukramJ** [reply]
Moin, testen sollte Problem los möglich sein. Die Aktualisierungen von HAHomematic werden jetzt auch sauber nachgeladen.

**Comment by itn3rd77** [comment]
Hallo Markus,

so habe die Einrichtung in der aktuellsten Version abgeschlossen. Hier mein erstes Feedback.

Heizungsgruppen werden nur bei IP richtig angelegt bei Gruppen mit alten RF Geräten sieht das aktuell in Home Assistant wie folgt aus:

![grafik](https://user-images.githubusercontent.com/2655169/143427572-f0990f1d-2550-4ab2-bee4-b4fec0cf2ad0.png)

Hier die Ansicht von der Gruppe auf der  CCU (RaspberryMatic):

![grafik](https://user-images.githubusercontent.com/2655169/143427725-7fd61467-e634-447b-839a-a2f1454a42d1.png)

Bei den Geräten in der Gruppe handelt es sich immer HM-CC-RT-DN.

Wenn Du weitere Infos benötigst, bitte melden.

Danke und Gruß
Ingo

**Reply by SukramJ** [reply]
Welche Version der Componente verwendest du? 
Wird der HM-CC-RT-DN einzeln richtig angezeigt?

**Reply by itn3rd77** [reply]
Der Test war mit der Version 597e121 und einzeln werden die HM-CC-RT-DN richtig angezeigt.

**Reply by SukramJ** [reply]
Gibt es Fehler im Log?

**Reply by itn3rd77** [reply]
In der 1c299b5 sieht es genauso aus (war aber zu erwarten) und Home Assistant habe ich die 2021.12.0.dev20211124 am laufen.

Hier auch nochmal ein Screenshot von einem der erkannten HM-CC-RT-DN 

![grafik](https://user-images.githubusercontent.com/2655169/143448683-8bc694db-4042-4484-acca-f9553ad78ec1.png)

**Reply by SukramJ** [reply]
Funktioniert die Bedienung über den HM-CC-RT-DN?

**Reply by itn3rd77** [reply]
Habe gerade nochmal einen Neustart mit neuem "sauberem" Logfile gemacht. Es gibt tatsächlich einen Fehler. Da es sich hier um eine Testinsanz hadeln kann ich gerne alles mögliche/unmögliche machen was Du brauchst. Sag einfach bescheid.

Hier das Logfile:
[home-assistant.log](https://github.com/danielperna84/hahomematic/files/7603254/home-assistant.log)

**Reply by itn3rd77** [reply]
> Funktioniert die Bedienung über den HM-CC-RT-DN?

Bedienung des HM-CC-RT-DN funktioniert.

**Reply by SukramJ** [reply]
Mach mal bitte ein Update. Den Fehler kenne ich, aber ich dachte immer die Testdaten sind defekt.

**Reply by SukramJ** [reply]
Hast du TLS aktiviert? Wäre noch nicht so gut. Hab ich noch nicht getestet.

**Reply by itn3rd77** [reply]
Ok etwas besser. Gruppe wird als Climate erkannt, aber nix geht oder wird angezeigt. Fehler gibt es dann z.B. bei Bedienung von Umstellen von Automatik auf Heizen.

![grafik](https://user-images.githubusercontent.com/2655169/143453949-fd787451-4c1a-46c7-b2fb-f0949c470e8c.png)

![grafik](https://user-images.githubusercontent.com/2655169/143453903-ce7d1593-e671-4df4-950d-ec21c605461c.png)

[home-assistant.log](https://github.com/danielperna84/hahomematic/files/7603457/home-assistant.log)

**Reply by itn3rd77** [reply]
> Hast du TLS aktiviert? Wäre noch nicht so gut. Hab ich noch nicht getestet.

Jupp TLS ist aktiviert.

**Reply by SukramJ** [reply]
Bitte TLS deaktivieren, da gibt es definitiv noch ein Problem.
Was die Gruppe anbelangt, weiss ich jetzt was zu tun ist. Dazu brauch ich aber ein neue Version von hahomematic.
Vielen Dank fürs Testen.

**Reply by SukramJ** [reply]
Jetzt sollten die HM Gruppen eigentlich funktionieren.

**Reply by SukramJ** [reply]
Die heutige Version unterstütz jetzt auch TLS richtig

**Reply by itn3rd77** [reply]
> Die heutige Version unterstütz jetzt auch TLS richtig

Perfekt! Funktioniert jetzt wie erwartet.

**Reply by SukramJ** [reply]
Sehr schön. Funktioniert der Rollandenaktor jetzt auch?

**Reply by itn3rd77** [reply]
> Sehr schön. Funktioniert der Rollandenaktor jetzt auch?

Hi, der Rollandenaktor HM-LC-Bl1PBU-FM funktiiniert nach dem neusten Update auch. Perfekt, danke!

**Comment by itn3rd77** [comment]
Feedback zu HM-LC-Bl1PBU-FM (Rollandenaktor):

![grafik](https://user-images.githubusercontent.com/2655169/143429463-41aceb38-74e8-4467-a38e-995556de54e2.png)

Wird zwar erkannt, aber funktioniert nichts. Hier mal der Trace von einem Klick auf runter:

```
Logger: hahomematic.entity
Source: /usr/local/lib/python3.9/site-packages/hahomematic/entity.py:295
First occurred: 11:51:33 (4 occurrences)
Last logged: 11:54:09

generic_entity: Failed to set state for: HM-LC-Bl1PBU-FM, LEQ0135383:1, LEVEL, 0
generic_entity: Failed to set state for: HM-LC-Bl1PBU-FM, LEQ0764339:1, LEVEL, 0
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/hahomematic/proxy.py", line 43, in __async_request
    return await self._executor_func(
  File "/usr/local/lib/python3.9/site-packages/hahomematic/client.py", line 191, in async_add_proxy_executor_job
    return await self.central.loop.run_in_executor(
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.9/xmlrpc/client.py", line 1458, in __request
    response = self.__transport.request(
  File "/usr/local/lib/python3.9/xmlrpc/client.py", line 1160, in request
    return self.single_request(host, handler, request_body, verbose)
  File "/usr/local/lib/python3.9/xmlrpc/client.py", line 1176, in single_request
    return self.parse_response(resp)
  File "/usr/local/lib/python3.9/xmlrpc/client.py", line 1348, in parse_response
    return u.close()
  File "/usr/local/lib/python3.9/xmlrpc/client.py", line 662, in close
    raise Fault(**self._stack[0])
xmlrpc.client.Fault: <Fault -1: 'Failure'>

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/hahomematic/entity.py", line 293, in send_value
    await self.proxy.setValue(self.address, self.parameter, value)
  File "/usr/local/lib/python3.9/site-packages/hahomematic/proxy.py", line 54, in __async_request
    raise ProxyException(ex) from ex
hahomematic.proxy.ProxyException: <Fault -1: 'Failure'>

```

**Reply by SukramJ** [reply]
Der Fehler sagt mir erst mal nichts. Hat vielleicht auch mit TLS zu tun.

**Reply by itn3rd77** [reply]
Auch nach dem Update mit TLS Fix bleiben die Fehler und die Stuerung funktioniert nicht:

```
Logger: hahomematic.entity
Source: /usr/local/lib/python3.9/site-packages/hahomematic/entity.py:248
First occurred: 09:01:19 (4 occurrences)
Last logged: 09:14:55

generic_entity: Failed to set state for: HM-LC-Bl1PBU-FM, LEQ0764350:1, LEVEL, 0
generic_entity: Failed to set state for: HM-LC-Bl1PBU-FM, LEQ0135383:1, LEVEL, 0
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/hahomematic/proxy.py", line 57, in __async_request
    return await self._executor_func(
  File "/usr/local/lib/python3.9/site-packages/hahomematic/client.py", line 195, in async_add_proxy_executor_job
    return await self.central.loop.run_in_executor(
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.9/xmlrpc/client.py", line 1458, in __request
    response = self.__transport.request(
  File "/usr/local/lib/python3.9/xmlrpc/client.py", line 1160, in request
    return self.single_request(host, handler, request_body, verbose)
  File "/usr/local/lib/python3.9/xmlrpc/client.py", line 1176, in single_request
    return self.parse_response(resp)
  File "/usr/local/lib/python3.9/xmlrpc/client.py", line 1348, in parse_response
    return u.close()
  File "/usr/local/lib/python3.9/xmlrpc/client.py", line 662, in close
    raise Fault(**self._stack[0])
xmlrpc.client.Fault: <Fault -1: 'Failure'>

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/hahomematic/entity.py", line 246, in send_value
    await self.proxy.setValue(self.address, self.parameter, value)
  File "/usr/local/lib/python3.9/site-packages/hahomematic/proxy.py", line 68, in __async_request
    raise ProxyException(ex) from ex
hahomematic.proxy.ProxyException: <Fault -1: 'Failure'>

```

**Reply by SukramJ** [reply]
Wenn ich das richtig sehe funktioniert das runterfahren nicht (LEQ0764350:1, LEVEL, 0)
Was passiert eigentlich wenn du den Slider bewegst?
Was passiert wenn auf Stop drückst?
